src/rt/cover.d

```diff
@@ -149,10 +149,80 @@ uint parseNum(const(char)[] s)
 T min(T)(T a, T b) { return a < b ? a : b; }
 T max(T)(T a, T b) { return b < a ? a : b; }
 
+version (unittest) __gshared bool inUnittest;
+
+string parseMergeOpt(string opt) @nogc nothrow
+{
+    switch (opt)
+    {
+        case "no", "No", "NO", "n", "N", "0":
+            merge = false;
+            return "0";
+        case "", "yes", "Yes", "YES", "y", "Y", "1":
+            merge = true;
+            return "1";
+        default:
+            if (!inUnittest)
+                fprintf(stderr, "DRT Error: Runtime option 'cov-merge' should "
+                        "be 1/y/yes or 0/n/no, got '%.*s'.\n",
+                        cast(int)opt.length, opt.ptr);
+            return "err";
+    }
+}
+
+unittest
+{
+    inUnittest = true;
+    scope (exit) inUnittest = false;
+
+    bool old_merge = merge;
+    scope (exit) merge = old_merge;
+
+    merge = true;
+    assert(parseMergeOpt("0") == "0" && !merge);
+    assert(parseMergeOpt("1") == "1" && merge);
+    assert(parseMergeOpt("n") == "0" && !merge);
+    assert(parseMergeOpt("y") == "1" && merge);
+    assert(parseMergeOpt("N") == "0" && !merge);
+    assert(parseMergeOpt("Y") == "1" && merge);
+    assert(parseMergeOpt("No") == "0" && !merge);
+    assert(parseMergeOpt("Yes") == "1" && merge);
+    assert(parseMergeOpt("NO") == "0" && !merge);
+    assert(parseMergeOpt("YES") == "1" && merge);
+    assert(parseMergeOpt("no") == "0" && !merge);
+    assert(parseMergeOpt("yes") == "1" && merge);
+    assert(parseMergeOpt("no") == "0" && !merge);
+    assert(parseMergeOpt("") == "1" && merge);
+    merge = false;
+    assert(parseMergeOpt("yEs") == "err" && !merge);
+    merge = true;
+    assert(parseMergeOpt("nO") == "err" && merge);
+    merge = false;
+    assert(parseMergeOpt("10") == "err" && !merge);
+    merge = true;
+    assert(parseMergeOpt("01") == "err" && merge);
+    assert(parseMergeOpt("xxx") == "err" && merge);
+    assert(parseMergeOpt("invalid") == "err" && merge);
+}
+
+void updateDRTOpts()
+{
+    import core.internal.traits : externDFunc;
+    alias rt_configCallBack = string delegate(string) @nogc nothrow;
+    alias fn_configOption = string function(string opt, scope rt_configCallBack dg, bool reverse) @nogc nothrow;
+    alias rt_configOption = externDFunc!("rt.config.rt_configOption", fn_configOption);
+
+    rt_configOption("covmerge", &parseMergeOpt);
+    rt_configOption("covsrcpath", (string s) { srcpath = s; });
+    rt_configOption("covdstpath", (string s) { dstpath = s; });
+}
+
 shared static ~this()
 {
     if (!gdata.length) return;
 
+    updateDRTOpts();
```

Comment:
Weird that `updateDRTOpts` is called in the desctructor, but I guess you don't need it before.

Reply:
Yeah, all is done in the destructor, and I thought adding a constructor to do the setup when it's never used before the destructor was adding too much complications for nothing (and adding a constructor means we need to care about ordering issues with the previous functions to set the options).